exporter/key_groups.go

```diff
@@ -173,7 +173,10 @@ for i=3,#ARGV do
   end
 end
 for i,key in ipairs(batch[2]) do
-  usage = redis.call("MEMORY", "USAGE", key)
+  local reply = redis.pcall("MEMORY", "USAGE", key)  
+  if type(reply) == "number" then
+    usage = reply;
+  end
```

Comment:
Question, should you exit if you don't get a value for usage? 
How will that get handled down in line 193/195 ?

Reply:
Hello,

local usage variable is initialized to 0 so no problem on line 193/195, is value will stay to 0 if MEMORY do not exists

The idea is to obtain at less key_groups count on redis version prior to 4.0, groups will always contain 0 for memory usage but the correct count.

A possibility to make better should be to not generate memory usage metrics when not available... but it's more complex and need more code modifications (but it is possible to do it ;-) ).

The current workaround make possible to use a limited key_groups scrapping that i think is better than the today error that generate nothing for key_group.

Reply:
That makes sense, thanks for the explanation!